### PR TITLE
[SYCL][Matrix] Fix checked matrix instructions

### PR DESCRIPTION
**Pull request by MrSidims**
There were incorrectly named and had incorrect operands.

See https://github.com/intel/llvm/pull/12497

### CODE REVIEW

File: sycl/include/CL/__spirv/spirv_ops.hpp [hunk 1]
```diff
@@ -51,33 +51,30 @@ template <typename T, typename Tp, std::size_t R, std::size_t C,
...
     __spv::__spirv_JointMatrixINTEL<Tp, R, C, L, S, U> *
-    __spirv_JointMatrixLoadCheckedINTEL(T *Ptr, std::size_t Stride,
-                                        size_t Height, size_t Width,
-                                        size_t CoordX, size_t CoordY,
-                                        __spv::MatrixLayout Layout = L,
-                                        __spv::Scope::Flag Sc = S,
-                                        int MemOperand = 0);
+    __spirv_CooperativeMatrixLoadCheckedINTEL(
+        T *Ptr, size_t CoordX, size_t CoordY, __spv::MatrixLayout Layout = L,
+        size_t Height = 0, size_t Width = 0, std::size_t Stride = 0,
```
Comment (dkhaldi) [inline sycl/include/CL/__spirv/spirv_ops.hpp:66]: do we need to set default values for these variables? It is not part of the spec, is it?

Comment (MrSidims) [reply sycl/include/CL/__spirv/spirv_ops.hpp:66]: Spec doesn't have them indeed. But SYCL headers implementation sets default value for layout, so following C++ rules any other value must be defaulted.

**Review by KseniyaTikhomirova — APPROVED** [review]
LGTM, please pay attention to tests

**Comment by MrSidims** [comment]
Warning comes from unused stride of checked fill API. There should be no such parameter, it also doesn't exist in the spec. I'll remove it in this PR.

**Review by dkhaldi — APPROVED** [review]
LGTM

File: sycl/include/sycl/ext/oneapi/matrix/matrix-intel.hpp [hunk 2]
```diff
@@ -612,22 +612,20 @@ template <typename Group, typename T, size_t NumRows, size_t NumCols, use Use,
...
-    const T2 &Value, size_t Stride, size_t Height, size_t Width, size_t CoordX,
+    const T2 &Value, size_t Height, size_t Width, size_t CoordX,
     size_t CoordY) {
 #if defined(__SYCL_DEVICE_ONLY__)
   using storage_element_type =
       typename oneapi::detail::jm_type_interpretation_helper_trait<
           T>::storage_element_type;
   Res.spvm = __spirv_CooperativeMatrixConstructCheckedINTEL<
-      storage_element_type, T, NumRows, NumCols,
+      storage_element_type, T, CoordX, CoordY, NumRows, NumCols,
```
Comment (ficol) [inline sycl/include/sycl/ext/oneapi/matrix/matrix-intel.hpp:622]:
```suggestion
      storage_element_type, T, NumRows, NumCols,
```
I think it shouldn't be added (these template args correspond to spirv_ops.hpp:48-51)

File: sycl/include/CL/__spirv/spirv_ops.hpp [hunk 3]
```diff
@@ -51,33 +51,30 @@ template <typename T, typename Tp, std::size_t R, std::size_t C,
...
           __spv::MatrixUse U,
           __spv::MatrixLayout L = __spv::MatrixLayout::RowMajor,
           __spv::Scope::Flag S = __spv::Scope::Flag::Subgroup>
-extern __DPCPP_SYCL_EXTERNAL void __spirv_JointMatrixStoreCheckedINTEL(
-    T *Ptr, __spv::__spirv_JointMatrixINTEL<Tp, R, C, L, S, U> *Object,
-    std::size_t Stride, size_t Height, size_t Width, size_t CoordX,
-    size_t CoordY, __spv::MatrixLayout Layout = L, __spv::Scope::Flag Sc = S,
-    int MemOperand = 0);
+extern __DPCPP_SYCL_EXTERNAL void __spirv_CooperativeMatrixStoreCheckedINTEL(
+    T *Ptr, size_t CoordX, size_t CoordY,
```
Comment (ficol) [inline sycl/include/CL/__spirv/spirv_ops.hpp:74]: Is it ok that CoordX, CoordY are size_t? In SPV_INTEL_joint_matrix.asciidoc, there is "X offset must be a scalar 32-bit integer type".

Comment (MrSidims) [reply sycl/include/CL/__spirv/spirv_ops.hpp]: It should be int32_t signed, it was fixed in another patch and after rebase it's brough here. Thanks!

File: sycl/include/sycl/ext/oneapi/matrix/matrix-intel.hpp (hunk 2 above)
Comment (MrSidims) [reply sycl/include/sycl/ext/oneapi/matrix/matrix-intel.hpp]: Indeed, thanks!

**Comment by MrSidims** [comment]
@intel/llvm-gatekeepers please help with the merge

**Pull request merged**